screens/NativeModule.tsx

```diff
@@ -1,20 +1,131 @@
-import React from 'react';
-import { StyleSheet, Text, Dimensions } from 'react-native';
+import React, { useState, useEffect, useCallback } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
 import { colors } from '../style/styleVariables';
 import ContainerFull from '../components/ContainerFull';
+import Header from '../components/Header';
+import { Camera } from 'expo-camera';
+import CameraComponent from '../components/CameraComponent';
+import MorseForm from '../components/MorseForm';
+import { translateToMorse } from '../morseMessenger/morseFunctions';
+
+const TORCH_ON = 'torch';
+const TORCH_OFF = '';
 
 export default function NativeModule() {
-  return (
-    <ContainerFull bgColor={colors.second}>
-      <Text style={styles.header}>Native Module</Text>
-    </ContainerFull>
-  );
-}
+  const [hasPermission, setHasPermission] = useState(null);
+  const [flashMode, setFlashMode] = useState('');
+
+  const buttonCallback = useCallback(input => {
+    asyncFlash(input);
+  }, []);
+
+  // Get the permision
+  useEffect(() => {
+    const effect = async () => {
+      const {
+        status
+      }: { status: string } = await Camera.requestPermissionsAsync();
+      setHasPermission(status === 'granted');
+    };
+    effect();
+  }, []);
+
+  function torchOffAfter(time: number) {
```

Comment:
it's usually nice to somehow indicate that the time is in ms without looking into implementation, e.g. `timeInMs: number`, same below